bitbots_quintic_walk/test/rostests/test_walk.py

```diff
@@ -0,0 +1,131 @@
+#!/usr/bin/env python3
+import time
+
+import rospy
+from bitbots_msgs.msg import JointCommand, SupportState
+from geometry_msgs.msg import Twist, Point, Pose, Quaternion
+
+from bitbots_test.mocks import MockSubscriber
+from bitbots_test.test_case import WebotsTestCase
+from nav_msgs.msg import Odometry
+
+
+class TestWalk(WebotsTestCase):
+
+    def test_start(self):
+        """ test if node starts correctly without warnings/errors/criticals
+        and if it is still there after some time (did not crash by itself)"""
+        # wait to make sure node is up
+        sub = MockSubscriber("DynamixelController/command", JointCommand, tcp_nodelay=True)
+        sub.wait_until_connected()
+        time.sleep(2)
+
+        self.assertNoNegativeRosLogs(node="walking")
+
+    def test_no_joint_goals(self):
+        """test if joint goals are published when walking is activated and only then"""
+        # setup
+        sub = MockSubscriber("DynamixelController/command", JointCommand, tcp_nodelay=True)
+        sub.wait_until_connected()
+
+        # execution
+        # wait some time
+        time.sleep(1)
+
+        # verification
+        sub.assertNothingReceived()
+
+    def test_joint_goals(self):
+        # setup
+        pub = rospy.Publisher("cmd_vel", Twist, queue_size=1)
+        sub = MockSubscriber("DynamixelController/command", JointCommand, tcp_nodelay=True)
+        sub.wait_until_connected()
+        # wait for support state to make sure the walking is started
+        sub_support_state = MockSubscriber("walk_support_state", SupportState, tcp_nodelay=True)
+        sub_support_state.wait_until_connected()
+        time.sleep(10)
+
+        # execution
+        cmd_vel = Twist()
+        cmd_vel.linear.x = 0.1
+        pub.publish(cmd_vel)
+        rospy.sleep(5)
+
+        # verification
+        # make sure something is published
+        sub.assertMessageReceived()
```

Comment:
You might want to use [`with_assertion_grace_period`](https://github.com/bit-bots/bitbots_tools/blob/6127589573e4d223ee315608b14383ee89029777/bitbots_test/src/bitbots_test/test_case.py#L23) to avoid sleeping for an unnecessarily long amount of time.